integration/run_test.go

```diff
@@ -171,3 +173,140 @@ func testWebServiceContainer(shell *shell.Shell, containerName string) {
 		"curl", "--retry", "5", "--retry-connrefused", "--retry-max-time", "30", "http://127.0.0.1",
 	)
 }
+
+type retryConfig struct {
+	// MaxRetries           int   `toml:"max_retries"`
+	// MinWaitMsec          int64 `toml:"min_wait_msec"`
+	// MaxWaitMsec          int64 `toml:"max_wait_msec"`
+	MaxRetries         int
+	MinWaitMsec        int64
+	MaxWaitMsec        int64
+	NetworkDisableMsec int64
+	ExpectedSuccess    bool
+}
+
+// TestNetworkRetry runs a container, disables network access to the remote image, asks the container
+// to do something requiring the remote image, waits for some/all requests to fail, enables the network,
+// confirms retries and success/failure
+func TestNetworkRetry(t *testing.T) {
+
+	const containerImage = alpineImage
+
+	tests := []struct {
+		name   string
+		config retryConfig
+	}{
+		// {
+		// 	name: "Fetch blob with no network interruption, no retries allowed",
+		// 	config: retryConfig{
+		// 		MaxRetries:         0,
+		// 		MinWaitMsec:        0,
+		// 		MaxWaitMsec:        0,
+		// 		NetworkDisableMsec: 0,
+		// 		ExpectedSuccess:    true,
+		// 	},
+		// },
+		// {
+		// 	name: "Fetch blob with 1s network interruption, no retries allowed",
+		// 	config: retryConfig{
+		// 		MaxRetries:         0,
+		// 		MinWaitMsec:        0,
+		// 		MaxWaitMsec:        0,
+		// 		NetworkDisableMsec: 1000,
+		// 		ExpectedSuccess:    false,
+		// 	},
+		// },
+		// {
+		// 	name: "Fetch blob with 1s network interruption, ~3-4s retries allowed",
+		// 	config: retryConfig{
+		// 		MaxRetries:         5,
+		// 		MinWaitMsec:        100,
+		// 		MaxWaitMsec:        1600,
+		// 		NetworkDisableMsec: 1000,
+		// 		ExpectedSuccess:    true,
+		// 	},
+		// },
+		// {
+		// 	name: "Fetch blob with 8s network interruption, ~3-4s retries allowed",
+		// 	config: retryConfig{
+		// 		MaxRetries:         5,
+		// 		MinWaitMsec:        100,
+		// 		MaxWaitMsec:        1600,
+		// 		NetworkDisableMsec: 8000,
+		// 		ExpectedSuccess:    false,
+		// 	},
+		// },
+		{
+			name: "Fake test for testing tests",
+			config: retryConfig{
+				MaxRetries:         2,
+				MinWaitMsec:        50,
+				MaxWaitMsec:        100,
+				NetworkDisableMsec: 2000,
+				ExpectedSuccess:    false,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			regConfig := newRegistryConfig()
+			sh, done := newShellWithRegistry(t, regConfig)
+			defer done()
+
+			registryHostIP := getIP(t, sh, regConfig.host)
+
+			config := `
+[blob]
+max_retries   = ` + strconv.Itoa(tt.config.MaxRetries) + `
+min_wait_msec = ` + strconv.FormatInt(tt.config.MinWaitMsec, 10) + `
+max_wait_msec = ` + strconv.FormatInt(tt.config.MaxWaitMsec, 10) + `
+
+[background_fetch]
+disable = true
+`
```

Comment:
I'll switch to using the existing tooling for serializing BlobConfig and BackgroundFetchConfig in a cleanup pass, after the tests work.